library/core/src/num/mod.rs

```diff
@@ -1581,3 +1585,138 @@ from_str_radix_size_impl! { i16 isize, u16 usize }
 from_str_radix_size_impl! { i32 isize, u32 usize }
 #[cfg(target_pointer_width = "64")]
 from_str_radix_size_impl! { i64 isize, u64 usize }
+
+#[unstable(feature = "kani", issue = "none")]
+mod verify {
```

Comment:
This code is quite repetitive. Could you try replace it with a macro? For example,

```rust
macro_rules! generate_unchecked_add_harness {
    ($type:ty) => {
        paste::paste! {
            #[kani::proof_for_contract(<$type>::unchecked_add)]
            pub fn [<check_unchecked_add_ $type>]() {
                let num1: $type = kani::any::<$type>();
                let num2: $type = kani::any::<$type>();

                unsafe {
                    num1.unchecked_add(num2);
                }
            }
        }
    };
}

// Use the macro to generate proof functions for all types.
generate_unchecked_add_harness!(i8);
generate_unchecked_add_harness!(i16);
generate_unchecked_add_harness!(i32);
generate_unchecked_add_harness!(i64);
generate_unchecked_add_harness!(i128);
generate_unchecked_add_harness!(isize);
generate_unchecked_add_harness!(u8);
generate_unchecked_add_harness!(u16);
generate_unchecked_add_harness!(u32);
generate_unchecked_add_harness!(u64);
generate_unchecked_add_harness!(u128);
generate_unchecked_add_harness!(usize);
```

Reply:
Thank you! This is a better way. I'll modify it.

Reply:
@feliperodri It looks like `paste` is a third-party crate. I  might need to add `paste = "1.0.15"` as a dependency to `Cargo.toml`. Am I allowed to do so? If so, where should I add this line? I tried `pub extern crate paste;` in `mod.rs` but it said "can't find crate".

Reply:
Please don't add dependencies to the standard library. You should be able to create a macro without the paste crate.

Reply:
For that, you need to add an extra argument to your harness that will be the name of the harness, and use that instead.

Reply:
Thank you! I tested with the following code without using `paste`. I was able to run the harness :)

```
    macro_rules! generate_unchecked_math_harness {
        ($type:ty, $method:ident, $harness_name:ident) => {

            #[kani::proof_for_contract($type::$method)]
            pub fn $harness_name() {
                let num1: $type = kani::any::<$type>();
                let num2: $type = kani::any::<$type>();

                unsafe {
                    num1.$method(num2);
                }
            }
        }
    }

    generate_unchecked_math_harness!(i8, unchecked_add, checked_unchecked_add_i8);
```

Reply:
@feliperodri  @celinval Our current approach is to create three macros:
* One for `add`, `sub`, `mul` - Require two integers for a single operation
* One for `shl` and `shr` - Require one integer and an u32 integer (bits to shift)
* One for `neg` - Require one integer

There might still be some repetitive code, but it is much less than before. I believe there is a better way to combine the three macros, but I have no idea how to do it at this moment. Does the current approach make sense?